internal/blocksync/pool.go

```diff
@@ -461,6 +460,25 @@ func (pool *BlockPool) pickIncrAvailablePeer(height int64) *bpPeer {
 		if height < peer.base || height > peer.height {
 			continue
 		}
+		// We only want to work with peers that are ready & connected (not dialing)
+		if pool.peerManager.State(nodeId) == "ready,connected" {
+			goodPeers = append(goodPeers, nodeId)
+		}
+
+		// Skip the ones with zero score to avoid connecting to bad peers
+		if pool.peerManager.Score(nodeId) <= 0 {
+			break
+		}
+	}
+
+	// randomly pick one
+	if len(goodPeers) > 0 {
+		rand.Seed(time.Now().UnixNano())
+		index := rand.Intn(len(goodPeers))
+		if index >= len(goodPeers) {
+			index = len(goodPeers) - 1
+		}
```

Comment:
nit - woudl this case ever be hit? could also use modulo to have more randomness

Reply:
I thought math.randn is inclusive?